### ISSUE TITLE
Allow return await statements

### ISSUE THREAD

**Issue by exanup** [leapfrogtechnology/tslint-config-leapfrog#5]
Allowing `return await [expression]` is not an anti-pattern anymore.

**TL;DR:** Always do return await when returning a promise to benefit full error stacktrace. If a function returns a promise, that function must be declared as async function and explicitly await the promise before returning it

**Otherwise:** The function that returns a promise without awaiting won't appear in the stacktrace. Such missing frames would probably complicate the understanding of the flow that leads to the error, especially if the cause of the abnormal behavior is inside of the missing function

Snipped explanation: 
> There is a number of excellent articles explained why return await should never be used outside of try block and even an ESLint rule that disallows it. The reason for that is the fact that since async/await become available with transpilers in Node.js 0.10 (and got native support in Node.js 7.6) and **until "zero-cost async stacktraces" was introduced in Node.js 10 and unflagged in Node.js 12**, return await was absolutely equivalent to return for any code outside of try block. It may still be the same for some other ES engines. This is why resolving promises before returning them is the best practice for Node.js and not for the EcmaScript in general

For more info: [Click here](https://github.com/goldbergyoni/nodebestpractices/blob/135bdf036686b5cf95d8a3bc99fa9dd3100a2026/README.md#-212-always-await-promises-before-returning-to-avoid-a-partial-stacktrace)

This line in particular seems to be enabling the rule: https://github.com/leapfrogtechnology/tslint-config-leapfrog/blob/17c4728fcd8d933c32a7fde6885c76c031285c17/tslint.json#L53
